fiftyone/core/dataset.py

```
@@ -9022,11 +9056,28 @@
 
-def _delete_dataset_doc(dataset_doc):
+def _handle_delete_generated_saved_view(dataset, view_doc):
+    if not any("_state" in s for s in view_doc.view_stages):
+        return
+
+    try:
+        # When deleting *generated* saved views, we mark the backing
+        # dataset as non-persistent so it can be garbage collected
+        view = dataset._load_saved_view_from_doc(view_doc)
+        view._dataset.persistent = False
+    except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)